lib/http2/connection.c

```diff
@@ -493,21 +492,34 @@ static int handle_incoming_request(h2o_http2_conn_t *conn, h2o_http2_stream_t *s
             return ret;
     }
 
-/* handle stream-level errors */
+    h2o_probe_log_request(&stream->req, stream->stream_id);
+
+    /* check existence of pseudo-headers */
+    if (h2o_memis(stream->req.input.method.base, stream->req.input.method.len, H2O_STRLIT("CONNECT"))) {
+        if (header_exists_map != (H2O_HPACK_PARSE_HEADERS_METHOD_EXISTS | H2O_HPACK_PARSE_HEADERS_AUTHORITY_EXISTS)) {
+            ret = H2O_HTTP2_ERROR_PROTOCOL;
+            goto SendRSTStream;
+        }
```

Comment:
Do we want to set `req->input.scheme`? I think we should, because there are a lot of places that depend on it being non-NULL.

Reply:
Fixed in ee13cb2.